plugins/CircuitViewer/io/CircuitLoader.cpp

```diff
@@ -129,74 +124,115 @@ struct CircuitProperties
+        boost::char_separator<char> separator(",");
+        boost::tokenizer<boost::char_separator<char>> tokens(targets,
+                                                             separator);
+        for_each(tokens.begin(), tokens.end(),
+                 [& targetList = targetList](const std::string& s) {
+                     targetList.push_back(s);
+                 });
+
+        boundingBox = Boxd(toVmmlVec(boundingBoxP0), toVmmlVec(boundingBoxP1));
     }
 
-    double circuitDensity = 0.0;
-    int32_t circuitRandomSeed = 0;
-    std::string circuitReport;
-    std::vector<std::string> circuitTargetsList;
-    std::string circuitTargets;
-    std::string circuitMeshFilenamePattern;
-    std::string circuitMeshFolder;
-    bool circuitUseSimulationModel = false;
-    int32_t circuitMeshTransformation = 0;
+    double density = 0.0;
+    int32_t randomSeed = 0;
+    std::string report;
+    std::vector<std::string> targetList;
+    std::string targets;
+    std::string meshFilenamePattern;
+    std::string meshFolder;
+    bool useSimulationModel = false;
+    bool transformMeshes = 0;
     ColorScheme colorScheme = ColorScheme::none;
 
-    double circuitStartSimulationTime = 0;
-    double circuitEndSimulationTime = 0;
-    double circuitSimulationStep = 0;
+    double startSimulationTime = 0;
+    double endSimulationTime = 0;
+    double simulationStep = 0;
 
     bool synchronousMode = false;
 
-    std::string loadCacheFile;
     GeometryQuality geometryQuality = GeometryQuality::high;
 
-    std::array<double, 3> circuitBoundingBoxP0;
-    std::array<double, 3> circuitBoundingBoxP1;
-
-    Boxd circuitBoundingBox;
+    std::array<double, 3> boundingBoxP0;
+    std::array<double, 3> boundingBoxP1;
+    Boxd boundingBox;
 };
 
+brain::GIDSet _getFilteredGIDs(const brain::Circuit& circuit,
+                               const std::string& target,
+                               const CircuitProperties& properties)
+{
+    const auto allGIDs = circuit.getRandomGIDs(properties.density / 100.0,
+                                               target, properties.randomSeed);
+
+    const auto& aabb = properties.boundingBox;
+    if (aabb.getSize() == Vector3f(0.f))
+        return allGIDs;
+
+    const Matrix4fs& transformations = circuit.getTransforms(allGIDs);
+    brain::GIDSet gids;
+    auto gid = allGIDs.begin();
+    for (size_t i = 0; i < transformations.size(); ++i, ++gid)
+        if (aabb.isIn(Vector3d(transformations[i].getTranslation())))
+            gids.insert(*gid);
+    return gids;
+}
+
+CompartmentReportPtr _openCompartmentReport(const brion::BlueConfig& blueConfig,
+                                            const std::string& name,
+                                            const brion::GIDSet& gids)
+{
+    try
+    {
+        auto source = blueConfig.getReportSource(name);
+        if (source.getPath().empty())
+        {
+            BRAYNS_ERROR << "Compartment report not found: " << name
```

Comment:
as mentioned by @karjonas, errors shall be thrown for the client to get them. Local prints can be done if you wish, but the throw is actually more important.

Reply:
Let me leave this code as is for this commit because it has changed in a follow up and I think that now it directly throws.